File: week09/week09-4.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        c = [0,0]
        for i in bills:
            if i==5:
                c[0]+=1
            if i==10:
                c[1]+=1
                c[0]-=1
            if i==20:
                if c[1]>=1:
                    c[1]-=1
                    c[0]-=1
                else:
                    c[0] -=3
            if c[0]<0 or c[1]<0:
                return False
        return True

File: week09/week09-4.cpp
class Solution {
public:
    bool lemonadeChange(vector<int>& bills) {
        int num[2]={};
        for(auto i:bills){
            if(num[0]<0||num[1]<0)return false;
            if(i==5)num[0]++;
            if(i==10){
                num[1]++;
                num[0]--;
            }
            if(i==20){
                if(num[1]>=1){
                    num[1]--;
                    num[0]--;
                }
                else num[0]-=3;
            }
        }
        if(num[0]<0||num[1]<0)return false;
        return true;
    }
};